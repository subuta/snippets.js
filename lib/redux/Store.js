import { build, format, snippets as s } from 'bld.js'
import _ from 'lodash'
import pluralize from 'pluralize'

export default () => {
  const imports = s.import([
    ['redux', null, [
      ['createStore'],
      ['applyMiddleware'],
      ['compose'],
    ]],
    ['redux-thunk', 'thunk'],
    ['src/utils/history', 'history'],
    ['react-router-redux', null, [
      ['routerMiddleware']
    ]],
    ['./modules', 'makeRootReducer'],
  ])

  return build`
    ${imports}
    
    const middlewares = [thunk, routerMiddleware(history)]
    
    const store = createStore(
      makeRootReducer(),
      compose(
        applyMiddleware(...middlewares),
        window.devToolsExtension ? window.devToolsExtension() : f => f
      )
    )
    
    store.injectedReducers = {};
    
    if (module.hot) {
      module.hot.accept('./modules', () => {
        const makeRootReducer = require('./modules').default
        store.replaceReducer(makeRootReducer(store.injectedReducers))
      })
    }
    
    export default store

  `
}