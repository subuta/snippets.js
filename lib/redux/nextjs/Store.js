import { build, format, snippets as s } from 'bld.js'
import _ from 'lodash'
import pluralize from 'pluralize'

export default () => {
  const imports = s.import([
    ['redux', null, [
      ['createStore'],
      ['applyMiddleware'],
      ['compose'],
    ]],
    ['redux-thunk', 'thunk'],
    ['./modules', 'makeRootReducer']
  ])

  return build`
    ${imports}
    
    const isBrowser = typeof window !== 'undefined'
    
    const middlewares = [thunk]
    
    let devtools = (f) => f
    if (isBrowser && window.devToolsExtension) {
      devtools = window.devToolsExtension()
    }
    
    // store.injectedReducers = {}
    
    // return createStore
    export default (initialState = {}) => {
      const store = createStore(
        makeRootReducer(),
        initialState,
        compose(
          applyMiddleware(...middlewares),
          // enable redux dev-tools
          isBrowser ? devtools : (f) => f
        ),
      )
    
      if (module.hot) {
        // Enable Webpack hot module replacement for reducers
        // https://github.com/reactjs/react-redux/releases/tag/v2.0.0
        if (module.hot) {
          module.hot.accept('./modules', () => {
            const makeRootReducer = require('./modules').default
            store.replaceReducer(makeRootReducer({}))
            // store.replaceReducer(makeRootReducer(store.injectedReducers))
          })
        }
      }
    
      return store
    }

  `
}
