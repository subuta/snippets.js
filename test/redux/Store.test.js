/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import Store from 'lib/redux/Store'

describe('redux/Store', () => {
  it('should create redux Store', () => {
    expect(format(Store())).toMatchSnapshot()
  })
})
