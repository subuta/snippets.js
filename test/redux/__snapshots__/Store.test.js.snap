// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux/Store should create redux Store 1`] = `
"import {createStore, applyMiddleware, compose} from 'redux'
import thunk from 'redux-thunk'
import history from 'src/utils/history'
import {routerMiddleware} from 'react-router-redux'
import makeRootReducer from './modules'

const middlewares = [thunk, routerMiddleware(history)]

const store = createStore(
  makeRootReducer(),
  compose(
    applyMiddleware(...middlewares),
    window.devToolsExtension ? window.devToolsExtension() : (f) => f
  )
)

store.injectedReducers = {}

if (module.hot) {
  module.hot.accept('./modules', () => {
    const makeRootReducer = require('./modules').default
    store.replaceReducer(makeRootReducer(store.injectedReducers))
  })
}

export default store
"
`;
